Hw2

awong-pdx opened the pull request:
### Code Changes I Made and Why
I added code to the files in the hw2 folder to complete the requirements for Homework 2 - CSS and JavaScript.
  
I didn't use some of the suggested resources in the assignment instructions, such as arrays and keydown events (Exercise 03), or window load events (Exercise 04).

***

### What Feedback I'm Looking For
Following the homework guidelines, I would appreciate any feedback on:
 - **Accessibility**
 - **Responsiveness**
 - **Airbnb style guide rules**
 - **Unnecessary code**
 - **Proper and consistent formatting**
 
I'd especially appreciate your thoughts on the areas where I might've deviated from what was expected, considering the suggested resources!
***

#### Thank you for your feedback!

awong-pdx committed:
Finish 01-flexbox.md

awong-pdx committed:
Finish 02-cssgrid.md

awong-pdx committed:
Rough functions for input word search, highlight

Started working on 03-Count and created
a function to highlight text and
another handleInput function.
Need to refine both functions

awong-pdx committed:
Update .gitignore

awong-pdx committed:
Updated highlight function

Added regex check to exclude partial word matches

awong-pdx committed:
Added styles for 03-Count

awong-pdx committed:
Tentatively finished 04-Background

Added styling with Bootstrap and
added several functions to handle
number input, random color generation,
start/stop background color change, and to edit the start/stop button.
Need to refine and add error handling.

awong-pdx committed:
Added styles to 05-Form

Used Bootstrap styles to format form to match assignment.
Added slight style and comment changes to 04-Background.

awong-pdx committed:
Added console log output for form submit

Implemented functions in 05-form.js to
capture form data and
output formatted data to console

awong-pdx committed:
Updated form handling to use form names

Added name attributes to all form elements
and used attributes in form capture/logging

awong-pdx committed:
Removed unnecessary comments

awong-pdx committed:
Minor code readability changes

huutribui commented on hw2/04-background.js:
```diff
@@ -0,0 +1,49 @@
+};
+
+const toggleButton = function toggleStartStopButton() {
+  if (startStop.className.includes('btn-primary')) {
```
I think it would be more clear if we check startStop innerText with "Start" or "Stop".

awong-pdx replied:
You're right, that's so much clearer. Thanks for the suggestion!

huutribui reviewed (commented):
Hi Alessandra,
I think you've done this homework perfectly. I only have 1 tiny suggestion but without that, your code is still running great.

awong-pdx committed:
Changed toggleButton to check for 'Start'

Also:
Renamed currentInterval to currentDelay and
changed colorGen() to allow for full range of colors

HaliTran commented on hw2/03-count.html:
```diff
@@ -0,0 +1,35 @@
+  <script defer src="03-count.js"></script>
+</head>
+
+<body class="container-xs bg-primary d-flex justify-content-center align-items-center vw-100 vh-100">
```
Using bootstrap classes on the individual elements (main, input div, etc) to center them allows for more flexibility down the line than putting it all on the body. This mainly only applies to bootstrap and not CSS since classes often work together (e.g container and rows). For example, if you later apply a class to a child of the body, it may or may not override the body style.

awong-pdx replied:
That's an interesting point! I'll move centering classes to `main`.

HaliTran commented on hw2/03-count.html:
```diff
@@ -0,0 +1,35 @@
+</head>
+
+<body class="container-xs bg-primary d-flex justify-content-center align-items-center vw-100 vh-100">
+  <main class="bg-light rounded p-4 h-75 w-75">
```
From my previous comment, I would recommend centering the main class by setting a width, then using margin auto. Reduces the amount of bootstrap classes needed! However, your approach works just as fine, just wanted to point this out!

HaliTran commented on hw2/05-form.js:
```diff
@@ -0,0 +1,32 @@
+
+const logForm = function captureAndLogRegistrationForm(event) {
+  const {
+    fullName, email, registrationStatus, coursesTaken, anythingElse,
```
Just wanted to say this is really nice! I didn't think about destructuring from the event.target object! It reduces a lot of redundant code.

HaliTran commented:
Hi Alessandra!

Your work for this homework was essentially flawless! I love how efficient and understandable your approach to many of the problems were! I only had a suggestion regarding your approach to styling the page with Bootstrap but that's it! Great job, I definitely learned how to improve my codebase by looking at yours, thank you!

awong-pdx committed:
Adjusted element centering with bootstrap classes

srimel commented on hw2/05-form.js:
```diff
@@ -0,0 +1,32 @@
+  const {
+    fullName, email, registrationStatus, coursesTaken, anythingElse,
+  } = event.target;
+  let coursesNamesTaken = [...coursesTaken];
```
I really like how you handled the coursesTaken data here, I always forget about the spread operator!

srimel commented on hw2/05-form.js:
```diff
@@ -0,0 +1,32 @@
+  registrationStatus.value !== ''
+    ? `Registration status: ${registrationStatus.value}`
+    : ''
+}
+  ${coursesNamesTaken.length > 0 ? `Courses taken: ${coursesNamesTaken}` : ''}
+  ${
+  anythingElse.value !== ''
+    ? `Additional Comments: ${anythingElse.value}`
+    : ''
+}
```
Very small style nit: Looks like the indentation is slightly out of alignment on lines 20, 26

awong-pdx replied:
Thanks for catching that!

srimel commented on hw2/04-background.js:
```diff
@@ -0,0 +1,51 @@
+  currentDelay = input.target.value * 1000;
+};
+
+const toggleButton = function toggleStartStopButton() {
+  if (startStop.textContent === 'Start') {
+    startStop.className = startStop.className.replace(
+      'btn-primary',
+      'btn-danger',
+    );
+    startStop.textContent = 'Stop';
+  } else {
+    startStop.className = startStop.className.replace(
+      'btn-danger',
+      'btn-primary',
+    );
+    startStop.textContent = 'Start';
+  }
```
I like how you used `replace` to switch the button styles! One suggestion I have for this is to extract out the logic that styles the button into a function. As it stands, it looks like you repeat the same button style logic twice which violates the DRY principle. It's definitely not a big deal in the context of this homework though, but if you were making a larger web app having that function would make the program more maintainable.

awong-pdx replied:
That's a great point! I'll look into separating that into another function.

srimel reviewed (approved):
Really good job with this assignment! Your code quality is very well designed and readable! I left a couple small suggestions on the PR

awong-pdx committed:
Fixed indentation and other style issues

awong-pdx committed:
Added check for min. interval, separated btn logic
